### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by meteorcloudy**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+python3 ./tools/add_module.py
+```
+
+The script will generate all require changes based on your input, please review, modify and commit the change, then send a PR to the BCR repository.
+
+If you are the project owner, you can set up the [Publish to BCR](https://github.com/apps/publish-to-bcr) Github App for your repository to automatically send a PR to the BCR when cutting a new release.
+
+## Presubmit
+
+Every module version must pass the BCR presubmit before getting merged. The presubmit validate the correctness and consistency of module information, then runs build and test targets specified in the `presubmit.yml` file. The BCR presubmit is driven by the [bcr_presubmit.py](https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/bazel-central-registry/bcr_presubmit.py) script on [Bazel CI](https://github.com/bazelbuild/continuous-integration/tree/master/buildkite#bazel-continuous-integration).
```
Comment (Wyverald) [inline docs/README.md:21]: the presubmit validate -> the presubmit validate**s**

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+- Verify the module version exists in the `metadata.json` of the module.
+- Verify the source archive URL matches the source repository specified in `metadata.json`.
+- Verify the source archive URL is stable if it comes from GitHub. (See [this discussion](https://github.com/bazel-contrib/SIG-rules-authors/issues/11#issuecomment-1029861300))
+- Verify the integrity values of the source archive and patch files (if any) are correct.
+- Verify the checked-in MODULE.bazel file matches the one in the extracted and patched source tree.
+- Check if the module is new or the `presubmit.yml` file is changed compared to the last version, if so a BCR maintainer review will be required to run jobs specified in `presubmit.yml`.
+
+Additional validations implemented in the [bcr_presubmit.py](https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/bazel-central-registry/bcr_presubmit.py) script:
+
+- No checked-in MODULE.bazel, source.json, patches files are not modified in the PR.
```
Comment (Wyverald) [inline docs/README.md:42]: Replace the first "No" with "The" since you have a "not" later; same on the next line

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+- No files outside of `modules/` directory are not modified in the pull request if the PR is adding a new module version.
+
+### Anonymous module test
+
+The modules in the BCR are meant to be used as a dependency of other Bazel modules. You can specify the targets you want to expose for your dependents in the `presubmit.yml` file, and the BCR presubmit will verify those targets can be built correctly when used as a dependency of a simple anonymous module.
```
Comment (Wyverald) [inline docs/README.md:47]: a dependency -> dependencies

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+For example, in `zlib`'s [metadata.json](https://github.com/bazelbuild/bazel-central-registry/blob/b842554be071a180d78c3a936c8ee0c94c74e8bd/modules/zlib/metadata.json#L17):
+
+```json
+"yanked_versions": {
+    "1.2.11": "CVE-2018-25032 (https://github.com/advisories/GHSA-jc36-42cf-vqwj)",
+    "1.2.12": "CVE-2022-37434 (https://github.com/advisories/GHSA-cfmr-vrgj-vqwv)"
+}
+```
+
+A Bzlmod user's build will start to fail if the yanked version is in the resolved dependency graph, the yanked reason will be presented in the error message. The user can choose to upgrade the dependency or they can bypass the check by specifying the `--allow_yanked_versions` flag or the `BZLMOD_ALLOW_YANKED_VERSIONS` environnement variable. Check [the documentation](https://bazel.build/reference/command-line-reference#flag--allow_yanked_versions) to learn how to use them.
```
Comment (Wyverald) [inline docs/README.md:120]: ", the yanked reason" -> ", and the yanked reason"

File: docs/README.md (hunk 1 above)
Comment (meteorcloudy) [reply docs/README.md]: Done

File: docs/README.md (hunk 2 above)
Comment (meteorcloudy) [reply docs/README.md]: Done

File: docs/README.md (hunk 3 above)
Comment (meteorcloudy) [reply docs/README.md]: Done

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    - '@zlib//:zlib'
+```
+
+In the presubmit, a simple anonymous module will be created with `MODULE.bazel`:
+
+```python
+bazel_dep(name="zlib", version="1.2.13")
+```
+
+Then the presubmit will verify building `@zlib//:zlib` succeed on all specified platforms.
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply docs/README.md]: Done

File: docs/README.md (hunk 4 above)
Comment (meteorcloudy) [reply docs/README.md]: Done

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    "1.2.11": "CVE-2018-25032 (https://github.com/advisories/GHSA-jc36-42cf-vqwj)",
+    "1.2.12": "CVE-2022-37434 (https://github.com/advisories/GHSA-cfmr-vrgj-vqwv)"
+}
+```
+
+A Bzlmod user's build will start to fail if the yanked version is in the resolved dependency graph, the yanked reason will be presented in the error message. The user can choose to upgrade the dependency or they can bypass the check by specifying the `--allow_yanked_versions` flag or the `BZLMOD_ALLOW_YANKED_VERSIONS` environnement variable. Check [the documentation](https://bazel.build/reference/command-line-reference#flag--allow_yanked_versions) to learn how to use them.
+
+## Versions format
+
+Bazel has a diverse ecosystem and projects using various versioning schemes, check [Bzlmod's version specification](https://bazel.build/build/bzlmod#version-format). If you need to update a module with only patch file changes, you can add `.bcr<N>` suffix to the version number.
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply docs/README.md]: Ah, nice catch!

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+While you can also specify `test_targets`, it may not always work since test targets can require additional dev dependencies that are not available when your project is not the root module.
+
+**Note that** the task config syntax follows [Bazel CI's specifications](https://github.com/bazelbuild/continuous-integration/tree/master/buildkite#configuring-a-pipeline).
+
+### Test module
+
+It's **highly recommended** to specify a test module that includes example usages of your module, which will help verify the basic APIs and functionalities of your module work correctly.
+
+A test module is located in a subdirectory of the extracted and patched source tree of the target module (the module you want to check in). You can specify the tasks in the `presubmit.yml` file under `bcr_test_module`.
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply docs/README.md]: Good idea, added! Also added an example for this.

**Review by meteorcloudy — COMMENTED** [review]
Thanks!

**Comment by meteorcloudy** [comment]
@philomathing Can you also take a look? ;)

**Comment by meteorcloudy** [comment]
@Wyverald Can you take a final look, I want to get this merged before the launch review today ;)

**Review by Wyverald — APPROVED** [review]
LGTM

**Pull request merged**